source/posix/thread.c

```diff
@@ -208,6 +210,23 @@ int aws_thread_launch(
         goto cleanup;
     }
 
+#if defined(USE_PTHREAD_SETAFFINITY)
```

Comment:
can we make these mutually exclusive, just in case?

Reply:
I would say the cmake check being hardened would be enough, but over time we're going to have a few builds that don't go through cmake and instead go through swift.package files and build.rs systems. Those will be emulating everything the cmake currently does and I'd like to prevent problems here if we get that part wrong.

Reply:
Thanks, I was wondering about this while adding these macros. I can see two possible approaches:

1. Add checks such as
#if (defined(X) && defined(Y)
#error
Probably simplest but would combinatorially increase if more mutually exclusive methods are added
2. Treat the method macros like an enum to be matched against possible values, e.g.
#if AWS_AFFINITY_METHOD == AWS_AFFINITY_METHOD_PTHREAD_ATTR

Reply:
I _think_ N is very small here, so I'm not particularly worried about 1. 2. is fine too. I leave it up to you, whatever seems cleanest and easiest to grok for maintainers, and still prevents a misconfiguration step from borking the builds.

Reply:
I ended up going with method 2, which is a bit more verbose but seems more self-documenting and maintainable.